src/Desktop.Analyzers/Core/DoNotCatchCorruptedStateExceptions.cs

```diff
@@ -29,132 +30,129 @@ public abstract class DoNotCatchCorruptedStateExceptionsAnalyzer<TLanguageKindEn
+        private bool ContainsHandleProcessCorruptedStateExceptionsAttribute(IMethodSymbol method, CompilationSecurityTypes compilationTypes)
         {
-            private readonly ISymbol _owningSymbol;
-            private readonly SyntaxNode _codeBlock;
-            private readonly Dictionary<TCatchClauseSyntax, ISymbol> _catchAllCatchClauses;
-
-            public abstract TLanguageKindEnum CatchClauseKind { get; }
-            public abstract TLanguageKindEnum ThrowStatementKind { get; }
-            protected CompilationSecurityTypes TypesOfInterest { get; private set; }
-            protected abstract ISymbol GetExceptionTypeSymbolFromCatchClause(TCatchClauseSyntax catchNode, SemanticModel model);
-            protected abstract bool IsThrowStatementWithNoArgument(TThrowStatementSyntax throwNode);
-            protected abstract bool IsCatchClause(SyntaxNode node);
-            protected abstract bool IslambdaExpression(SyntaxNode node);
-
-            protected CodeBlockAnalyzer(CompilationSecurityTypes compilationTypes, ISymbol owningSymbol, SyntaxNode codeBlock)
+            ImmutableArray<AttributeData> attributes = method.GetAttributes();
+            return attributes.Any(
+                attribute => attribute.AttributeClass.Equals(compilationTypes.HandleProcessCorruptedStateExceptionsAttribute));
+        }
+
+        /// <summary>
+        /// Walks an IOperation tree to find catch blocks that contain no empty throw statements.
+        /// </summary>
+        private class CatchInsideBlockWalker : OperationWalker
+        {
+            private readonly CompilationSecurityTypes _compilationTypes;
+
+            public IList<ICatchClause> CatchAllCatchClauses { get; } = new List<ICatchClause>();
+
+            public CatchInsideBlockWalker(CompilationSecurityTypes compilationTypes)
             {
-                _owningSymbol = owningSymbol;
-                _codeBlock = codeBlock;
-                _catchAllCatchClauses = new Dictionary<TCatchClauseSyntax, ISymbol>();
-                TypesOfInterest = compilationTypes;
+                _compilationTypes = compilationTypes;
             }
 
-            public void AnalyzeCatchClause(SyntaxNodeAnalysisContext context)
+            public override void VisitCatch(ICatchClause operation)
             {
-                var catchNode = (TCatchClauseSyntax)context.Node;
-                ISymbol exceptionTypeSymbol = GetExceptionTypeSymbolFromCatchClause(catchNode, context.SemanticModel);
+                Visit(operation.Filter);
 
-                if (IsCatchTypeTooGeneral(exceptionTypeSymbol))
+                if (IsCaughtTypeTooGeneral(operation.CaughtType))
                 {
-                    SyntaxNode parentNode = catchNode.Parent;
-                    while (parentNode != _codeBlock)
+                    // TODO: Abort in case parent is a lambda.
+                    // for now there doesn't seem to be any way to annotate lambdas with attributes
```

Comment:
@genlu It looks like you are the author of this analyzer. This line originally came from https://github.com/dotnet/roslyn/pull/3954/files#diff-163597667eb0517f0d40f26a13176842R112. I'm rewriting to use `IOperation`, but not sure what this comment means. Can you come up with a unittest that shows how it is supposed to work?